File: Monocle/Api/Enums.cs
using SDG.Unturned;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Monocle.Api
{
    public enum AuthorizedUserType
    {
        Observer,
        Administrator
    }

    enum ChatMode
    {
        Global,
        Local,
        Group,
        Say,
        Welcome,
    }

    enum VehicleType
    {
        Car,
        Plane,
        Helicopter,
        Blimp,
        Boat,
        Train,
    }

    enum ItemType
    {
        Hat,
        Pants,
        Shirt,
        Mask,
        Backpack,
        Vest,
        Glasses,
        Gun,
        Sight,
        Tactical,
        Grip,
        Barrel,
        Magazine,
        Food,
        Water,
        Medical,
        Melee,
        Fuel,
        Tool,
        Barricade,
        Storage,
        Beacon,
        Farm,
        Trap,
        Structure,
        Supply,
        Throwable,
        Grower,
        Optic,
        Refill,
        Fisher,
        Cloud,
        Map,
        Key,
        Box,
        ArrestStart,
        ArrestEnd,
        Tank,
        Generator,
        Detonator,
        Charge,
        Library,
        Filter,
        Sentry,
        VehicleRepairTool,
        Tire,
        Compass,
        OilPump
    }

    enum ItemRarity
    {
        Common,
        Uncommon,
        Rare,
        Epic,
        Legendary,
        Mythical
    }
}

File: MonocleFrontend/src/app/types/enums.ts
export enum ErrorType
{
    InvalidRequestType = "InvalidRequestType",
    UserNotFound = "UserNotFound",
    InvalidRequestData = "InvalidRequestData",
}

export enum RequestType
{
    Authenticate = "Authenticate",
    Players = "Players",
    PlayerDetails = "PlayerDetails",
    Structures = "Structures",
    Barricades = "Barricades",
    Vehicles = "Vehicles",
    ServerInfo = "ServerInfo",
    PlayerScreenshot = "PlayerScreenshot",
}

export enum ServerResponseType
{
    Players = "Players",
    CurrentBuildings = "CurrentBuildings",
    SuccessfulLogin = "SuccessfulLogin",
    Vehicles = "Vehicles",
    Barricades = "Barricades",
    Structures = "Structures",
    ServerInfo = "ServerInfo",
    PlayerScreenshot = "PlayerScreenshot",
}

export enum EventType
{
    PlayerDeath = "PlayerDeath",
    PlayerMessage = "PlayerMessage",
    PlayerJoined = "PlayerJoined",
    PlayerLeft = "PlayerLeft",
}

export enum ChatMode {
    Local = 1,
    Global,
    Group,
    Say,
    Welcome
}

export enum VehicleType {
    Car = 1,
    Plane,
    Helicopter,
    Blimp,
    Boat,
    Train,
}

export enum ItemRarity
{
    Common,
    Uncommon,
    Rare,
    Epic,
    Legendary,
    Mythical
}